### PR TITLE
NEW TASK PROPOSAL : Advanced Task : Documentation Engine

### PR DESCRIPTION
**Pull request by YashKumarVerma**
Adding a new task
---
https://codein.withgoogle.com/dashboard/tasks/6448508355739648/

---

# Labyrinth: Enhance the Documentation Engine (Advanced Task)

The [FOSSASIA Labyrinth](https://github.com/fossasia/labyrinth/) allows you to contribute parts to a huge labyrinth.
To make the game development fast and more easy, build the documentation engine of the game / enhance the features/ add a new feature to the documentation engine.
The Documentation engine is proposed to render the documentation from static files onto a webpage backed by javascript.

Requirements
- Build the documentation engine 
- If the engine already exists, add a feature to it. (pagination/search etc)
- Write Documentation about any one API of the game and add it to the engine

Expected Outcome:
- a documentation engine which renders docs dynamically (with a new feature)
- a new API documentation added

Hints:
- If someone wants to work with you, let them do that. You can use techniques like [pair programming](https://www.youtube.com/watch?v=vgkahOzFH2Q) or collaborate together on branches of your forks. You can attribute a pull-request also to other people who helped you out or submit a pull-request where you and an other person contributed together.
- You can join our [chat on gitter](https://gitter.im/fossasia/labyrinth) if you get stuck to get help or just hang out and have a nice time with others.

---

Categories:
- [X] Code
- [X] User Interface
- [X] Documentation / Training
- [X] Quality Assurance
- [ ] Outreach / Research

### CODE REVIEW

**Comment by YashKumarVerma** [comment]
@niccokunzmann

**Comment by niccokunzmann** [comment]
I would like to formulate an answer where I take time to collect my thoughts on the tasks. So, if you could wait until I am satisfied with my answer, I would appreciate this.

**Comment by YashKumarVerma** [comment]
@niccokunzmann 
The overview is to create a dynamic documentation which loads data from files (which can be added by students).
This serves:
1. The Main code remains integral. No changes to that
2. Adding / Maintaining things is easy
3. Nothing  breaks 

and since our game keeps on developing so fast, changing an html documentation will be a messy thing. adding separate `.md` for each thing will be difficult to search.

**Comment by niccokunzmann** [comment]
I will merge it but I am not convinced that I would post the task. I try to explain here why from different points of view. First of all I want to say that I value your intention behind the task and I think, what you have in mind improves the game and the development.


- GCI
  - see docs before reading further: https://github.com/fossasia/labyrinth/blob/master/tasks/README.md
  - what is a dynamic documentation engine?
  - What if - and they will - students just add a documentation engine or claim an html page rendered from markdown is dynamic, if they add something new?
  - In GCI I need to be objective
    - "a documentation engine which renders docs dynamically (with a new feature)" this can only be accepted when the prior work is done. After coding on the engine, we will have a total different meaning of the task.
      - where do students synchronize - what if 3 jump on it and all add a different engine? I will accept that.
    - there are existing engines. Should we create a new one?

So, there is this task
- adding a new feature https://github.com/fossasia/labyrinth/blob/master/tasks/add-a-new-feature.md
- enable a GCI task https://github.com/fossasia/labyrinth/blob/master/tasks/enable-a-new-gci-task.md
- adding documentation https://github.com/fossasia/labyrinth/blob/master/tasks/document-how-to-contribute.md

the first one can be used to e.g. add read the docs  
the second one would be the one which would enable a documentation used by the engine  
the third one is there and can be claimed. Maybe we should improve it.

Summing up, I see these as your concerns and I would like to help you:
- having a documentation engine
  - getting help from others - a task as a means for that
  - good documentation for the game
- new accepted gci task instances

What do you think? I wrote a lot I guess..

**Comment by YashKumarVerma** [comment]
Whoa 😲 
I got what you wanted to state. Completely. I think we can add an issue instread of a task for this. My concern for adding a task for it was, so that fellow students participate in it in good number.

If the same is achieved through an issue, i'm way more happy than that. 😄
